src/python/pants/option/arg_splitter_test.py

```diff
@@ -293,69 +298,82 @@ def help_no_arguments_test(command_line: str, **expected):
     "command_line, expected",
     [
         help_no_arguments_test("./pants"),
-        help_no_arguments_test("./pants help"),
-        help_no_arguments_test("./pants -h"),
-        help_no_arguments_test("./pants --help"),
-        help_no_arguments_test("./pants help-advanced", expected_help_advanced=True),
-        help_no_arguments_test("./pants help --help-advanced", expected_help_advanced=True),
-        help_no_arguments_test("./pants --help-advanced", expected_help_advanced=True),
-        help_no_arguments_test("./pants --help --help-advanced", expected_help_advanced=True),
-        help_no_arguments_test("./pants --help-advanced --help", expected_help_advanced=True),
-        help_no_arguments_test("./pants help-all", expected_help_all=True),
+        help_no_arguments_test("./pants help", "help"),
+        help_no_arguments_test("./pants -h", "help"),
+        help_no_arguments_test("./pants --help", "help"),
+        help_no_arguments_test(
+            "./pants help-advanced", "help-advanced", expected_help_advanced=True
+        ),
+        help_no_arguments_test(
+            "./pants --help-advanced", "help-advanced", expected_help_advanced=True
+        ),
+        help_no_arguments_test("./pants help-all", "help-all", expected_help_all=True),
         help_test(
-            "./pants -f",
-            expected_goals=[],
-            expected_scope_to_flags={"": ["-f"]},
+            "./pants --help-advanced --help",
+            expected_goals=["help"],
+            expected_scope_to_flags={"": [], "help": [], "help-advanced": []},
             expected_specs=[],
+            expected_help_advanced=True,
         ),
         help_test(
-            "./pants help check -x",
-            expected_goals=["check"],
-            expected_scope_to_flags={"": [], "check": ["-x"]},
+            "./pants --help --help-advanced --builtin-option --help-advanced-option",
+            expected_goals=["help-advanced"],
+            expected_scope_to_flags={
+                "": [],
+                "help": ["--builtin-option"],
+                "help-advanced": ["--option"],
+            },
+            expected_specs=[],
+            expected_help_advanced=False,
+        ),
+        help_test(
+            "./pants -f",
+            expected_goals=[],
+            expected_scope_to_flags={"": ["-f"]},
             expected_specs=[],
         ),
         help_test(
             "./pants help check -x",
             expected_goals=["check"],
-            expected_scope_to_flags={"": [], "check": ["-x"]},
+            expected_scope_to_flags={"": [], "help": ["-x"], "check": []},
```

Comment:
My reasoning here is that I see `./pants help check -x` as `./pants help <thing[s]...> <help option[s]...>`
As shown in a previous test, you can provide options for the thing you ask help on, by including its scope:
```
./pants help check --check-x
```